### PR TITLE
Float8E4M3FNUZ -> Float8E4M3FN for NVIDIA PTX

### PR DESCRIPTION
**Pull request by acollins3**
Fix MLIR type used for e4m3 fp8 type in NVIDIA PTX codegen.

https://github.com/triton-lang/triton/issues/3681

### CODE REVIEW

File: python/src/ir.cc [hunk 1]
```diff
@@ -748,7 +748,7 @@ void init_triton_ir(py::module &&m) {
            // TODO: fp8e4nv is using Float8E4M3FNUZType, which
```
Comment (chsigg) [inline python/src/ir.cc:748]: Does this comment still apply?

File: include/triton/Dialect/Triton/IR/TritonTypes.td [hunk 2]
```diff
@@ -15,7 +15,7 @@ class TritonTypeDef<string name, string _mnemonic, list<Trait> traits = []>
 }
 
 // Floating-point Type
-def TT_Float : AnyTypeOf<[F8E4M3FNUZ, F8E5M2, F8E5M2FNUZ, F16, BF16, F32, F64], "floating-point">;
+def TT_Float : AnyTypeOf<[F8E4M3FN, F8E4M3FNUZ, F8E5M2, F8E5M2FNUZ, F16, BF16, F32, F64], "floating-point">;
```
Comment (chsigg) [inline include/triton/Dialect/Triton/IR/TritonTypes.td:18]: Should `F8E4M3FNUZ` really need to be removed?

Also below, in some places `F8E4M3FN` got added, in other places `F8E4M3FNUZ` is being replaced. It would be good to explain what we want in the PR description and apply it consistently. Or maybe I'm missing something and this is all intentional?

File: lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp [hunk 3]
```diff
@@ -156,16 +156,16 @@ static Value getSharedMemoryMMAOperand(Value v, mlir::PatternRewriter &rewriter,
 
   // LocalAllocOp lowering doesn't support going from DotOperandEncoding
   // to SharedEncoding.
-  if (auto dotOpEnc = mlir::dyn_cast<DotOperandEncodingAttr>(
-          argType.getEncoding())) {
+  if (auto dotOpEnc =
```
Comment (chsigg) [inline lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp:159]: Please undo unrelated formatting changes. Also below.

File: python/src/ir.cc (hunk 1 above)
Comment (acollins3) [reply python/src/ir.cc:748]: Well spotted. I'll remove the comment

**Comment by chsigg** [comment]
This PR has landed [upstream](https://github.com/triton-lang/triton/commit/9665bc0cbe37e2ffe72f6a3b8b09400a9b733c76), closing.
